R/get_draws_df.R

```diff
@@ -5,38 +5,83 @@
 #' and the 3 relevant mappings from stan indices to the real data, in order
 #' to generate a dataframe containing the posterior draws of the counts (e.g.
 #' hospital admissions), the wastewater concentration values, the "global" R(t),
-#' and the "local" R(t) estimates + the critical metadata in the data
+#' and the "local" R(t) estimates + the critical metadata in the data.
+#' This funtion has a default method that takes the two sets of input data,
+#' the last of stan arguments, and the CmdStan fitting object, as well as an S3
+#' method for objects of class 'wwinference_fit'
 #'
 #'
-#' @param ww_data A dataframe of the preprocessed wastewater concentration data
-#' used to fit the model
+#' @param x Either a dataframe of wastewater observations, or an object of
+#' class wwinference_fit
 #' @param count_data A dataframe of the preprocessed daily count data (e.g.
 #' hospital admissions) from the "global" population
+#' @param stan_args A list containing all the data passed to stan for fitting
+#' the model
 #' @param fit_obj a CmdStan object that is the output of fitting the model to
-#' the `ww_data` and `count_data`
-#' @param date_time_spine A tibble mapping the time index in stan (observed +
-#' nowcast + forecast) to real dates
-#' @param lab_site_spine A tibble mapping the site-lab index in stan to the
-#' corresponding site, lab, and site population
-#' @param subpop_spine A tibble mapping the site index in stan to the
-#' corresponding subpopulation (either a site or the auxiliary site we add to
-#' represent the rest of the population)
-#'
+#' `x` and `count_data`
+#' @param ... additional arguments
 #' @return  A tibble containing the full set of posterior draws of the
 #' estimated, nowcasted, and forecasted: counts, site-level wastewater
 #' concentrations, "global"(e.g. state) R(t) estimate, and the  "local" (site +
 #' the one auxiliary subpopulation) R(t) estimates. In the instance where there
 #' are observations, the data will be joined to each draw of the predicted
 #' observation to facilitate plotting.
 #' @export
-get_draws_df <- function(ww_data,
-                         count_data,
-                         fit_obj,
-                         date_time_spine,
-                         lab_site_spine,
-                         subpop_spine) {
+get_draws_df <- function(x, ...) {
+  UseMethod("get_draws_df")
+}
+
+#' S3 method for extracting posterior draws alongside data for a
+#' wwinference_fit object
+#'
+#' This method overloads the generic get_draws_df function specifically
+#' for objects of type 'wwinference_fit'.
+#'
+#' @rdname get_draws_df
+#' @export
+get_draws_df.wwinference_fit <- function(x, ...) {
+  get_draws_df.default(
+    x = x$input_data$input_ww_data,
+    count_data = x$input_data$input_count_data,
+    stan_args = x$stan_args,
+    fit_obj = x$fit
+  )
+}
+
+#' @rdname get_draws_df
+#' @export
+get_draws_df.default <- function(x,
```

Comment:
This is if someone wants to pass in the tibbles themselves rather than the `wwinference` class object

Reply:
I guess we could get rid of this and parse the args from the `wwinference` object but I think I have a slight preference for the default being that we pass in the individual components of rather than a class object.

Reply:
then why isn't that a .dataframe method?

Reply:
The way that this is currrently coded literally anything people try and input gets passed to .default. If you instead had .default be say empty you could avoid some fairly large amount of input checks (as the message can be in .default)

Reply:
all of this aside for this actual function it seems like expanding  on posterior / on of the other standard names from stan land

Reply:
I think the difference between this and `posterior::as_draws_df()` is that this just returns a few of the variables (observables + R(t) estimates) bound to dates/data if available, using `tidybayes()`. Perhaps I should rename to something more descriptive re `extract_posterior_draws_w_data()` ? Open to changing things around as now is the time to nail this down! But I do think this table, or something similar, is a useful output since this is what someone would use for plotting, or computing quantiles, etc.

Reply:
> then why isn't that a .dataframe method?
 Bc I didn't know this was a thing. 

To recap your suggestion: 
1. change default to be empty (so just `...`?)
2. add a `.dataframe` method that takes in the necessary dataframes, currently described in `.default`
3. add a message in `.default()` telling the user to either pass the required dataframe args or the `wwinference_fit` object?

Reply:
Default method now fails (stop) with message indicating the available methods.